### PR TITLE
Support to n plots using lists

### PR DESCRIPTION
**Pull request by EliosMolina**
I suggest not to use `**kwargs`, I think it introduces redundancy. Everything could be in parameters dictionaries. I leave a TODO in `update_params()`.

### CODE REVIEW

File: sdb/stocksdashboard.py [hunk 1]
```diff
@@ -206,17 +207,16 @@ def plot_stock(self, input_data=None, p=None, column='adj_close',
...
         p_to_hover = []
         for i, stock in enumerate(data):
             x, y = self.get_x_y(stock, column)
             __params = self.get_params(params, names[i], colors[i])
             __p = p.line(x, y, **__params)
             p_to_hover.append(__p)
 
-        assert(p_to_hover == len(data)), "Number of Lines " + \
-                                         "don't match data dimension."
+        assert(len(p_to_hover) == len(data)), "Number of Lines " + \
```
Comment (mabelvj) [inline sdb/stocksdashboard.py]: Well done picking this up!! 👍

File: sdb/stocksdashboard.py [hunk 2]
```diff
@@ -227,21 +227,20 @@ def plot_stock(self, input_data=None, p=None, column='adj_close',
...
-
-        # open a browser
-        layout = gridplot([p1, p2],
-                          plot_width=self.width, plot_height=self.height,
+    def build_dashboard(self,
+                        data_list=[],
+                        params_list=[],
+                        title="stocks.py example"):
+        plots = []
+        for data, params in zip(data_list, params_list):
```
Comment (mabelvj) [inline sdb/stocksdashboard.py]: I have to check this out, but we can't have infinite vertical plots due to the limitation of the screen.

File: sdb/stocksdashboard.py [hunk 3]
```diff
@@ -206,17 +207,16 @@ def plot_stock(self, input_data=None, p=None, column='adj_close',
 
         data, names = Formatter().format_data(input_data)
         colors = get_colors(len(data))
-        params = self.update_params(params, kwargs, names)
-
+        params = self.update_params(params=params, names=names)
```
Comment (mabelvj) [inline sdb/stocksdashboard.py]: I don't agree with this change, I think we should be able to setup a set of parameters in the building of the dashboard aside from the particular aesthetics.

File: sdb/stocksdashboard.py [hunk 4]
```diff
@@ -195,7 +196,7 @@ def get_params(self, params, name, color):
 
     def plot_stock(self, input_data=None, p=None, column='adj_close',
                    title="Stock Closing Prices", ylabel='Price',
-                   add_hover=True, params={}, **kwargs):
+                   add_hover=True, params={}):
```
Comment (mabelvj) [inline sdb/stocksdashboard.py]: I think we should be able to give additional params to the plots that are general to all the plots.

File: sdb/main.py [hunk 5]
```diff
@@ -16,18 +16,13 @@
...
-StocksDashboard().build_dashboard(data1={'AAPL': AAPL, 'GOOG': GOOG,
-                                         'IBM': IBM, 'MSFT': MSFT},
-                                  data2={'AAPL_avg': aapl_avg},
-                                  params={'GOOG': {'line_dash': 'dashed'},
-                                          'AAPL': {'color': 'blue'}},
-                                  params2={'color': 'orange',
-                                           'line_width': 1.5},
-                                  line_width=2)
+StocksDashboard().build_dashboard(
+    data_list=[{'AAPL': AAPL, 'GOOG': GOOG,
```
Comment (mabelvj) [inline sdb/main.py]: I think this is ok, though maybe overcomplicating the problem. Maybe a dict would be better and we could add that as title for the plot.

**Comment by EliosMolina** [comment]
Ok if we are going to use the `**kwargs`, then we have to include them again in the rest of functions.

**Comment by mabelvj** [comment]
Added underscores for all private methods and variables. Now only methods accessible by the user have no underscore.

**Comment by mabelvj** [comment]
Added tests for the new methods: `Formatter().format_input_data`, and `Formatter().format_params`, which format input data passed to `StocksDashboard().build_dashboard()` to check they are dict or list of params per plot.

**Comment by mabelvj** [comment]
Merging

**Pull request merged**